src/main/java/org/folio/rest/support/ShelvingOrderUpdate.java

```diff
@@ -0,0 +1,197 @@
+package org.folio.rest.support;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.pgclient.PgConnection;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.Tuple;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dbschema.Versioned;
+import org.folio.rest.jaxrs.model.Item;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.persist.PgUtil;
+import org.folio.rest.persist.PostgresClient;
+
+public class ShelvingOrderUpdate {
+
+  private static final Logger log = LogManager.getLogger();
+
+  private static final String SQL_EXISTS_ITEMS_TO_UPDATE = new StringBuilder()
+      .append("SELECT EXISTS (SELECT null FROM item WHERE not (jsonb ? 'effectiveShelvingOrder'))")
+      .toString();
+
+  private static final String SQL_SELECT_ITEMS = new StringBuilder()
+      .append("SELECT id AS \"id\", ")
+      .append("jsonb AS \"itemJson\" ")
+      .append("FROM item WHERE not (jsonb ? 'effectiveShelvingOrder') ")
+      .append(" FOR UPDATE ")
+      .append("LIMIT $1").toString();
+
+
+  private static final String SQL_UPDATE_ITEMS = new StringBuilder()
+      .append("UPDATE ITEM ")
+      .append("SET jsonb = $1 ")
+      .append("WHERE id = $2").toString();
+
+  // Defines version threshold where this property was introduces from
+  private static final String FROM_VERSION_SHELVING_ORDER = "20.1.0";
+
+  private static final String OPERATION_STATUS_VERSION_MISMATCH = "Module version isn't expected";
+  private static final String OPERATION_STATUS_ITEMS_UPDATES_SUCCESSFUL = "Items updates completed successfully";
+  private static final String OPERATION_STATUS_ITEMS_UPDATES_FAILED = "Items updates failed";
+
+  private Versioned versionChecker;
+
+  private int updateBunchSize;
+  private Map<UUID, String> itemsShelvingOrderMap;
+  private Map<UUID, Item> itemsMap;
+
+  public ShelvingOrderUpdate(int updateBunchSize) {
+    this.versionChecker = new Versioned() { };
+    this.versionChecker.setFromModuleVersion(FROM_VERSION_SHELVING_ORDER);
+    this.updateBunchSize = updateBunchSize;
+
+    itemsShelvingOrderMap = new LinkedHashMap<>();
+    itemsMap = new LinkedHashMap<>();
+  }
+
+  public Future<String> updateItems(TenantAttributes attributes, Map<String, String> headers, Context vertxContext) {
+    Promise<String> promise = Promise.promise();
+
+    String fromModuleVersion = attributes.getModuleFrom().trim();
+    if (StringUtils.isBlank(fromModuleVersion) || versionChecker.isNewForThisInstall(fromModuleVersion)) {
+      // Update items only for versions before the expected
```

Comment:
It's very modern with these isBlank thingy.. Okapi will NEVER send a fromModuleVersion with a blank. It will be omitted and for YOU == null.

Reply:
Replacing of isBlank from the "Apache's common lang" with just checking for null (possibly by no less popular Objects.isNull) will avoid of using extra dependencies here. Because isBlank works well with all the undesired strings too. But if we are pretty sure about some of handling from Okapi side, it make sense to perform only bare minimum checking to not be confused with varieties of expected values of course.

Reply:
Well.. That's the key "undesired" string. Are they desired? No. That's why systems should not allow it - EXCEPT when it's coming from a user and even there is should be thrown away (eg passwords). But not a serious issue.

Reply:
I see, thank you for clarifieng.